testsuite/test_tagimport.py

```diff
@@ -0,0 +1,89 @@
+import pytest
+import csv
+from snoop.data import collections, models
+from conftest import TESTDATA
+from django.conf import settings
+from django.core.management import call_command
+import requests
+import time
+
+ES_URL = settings.SNOOP_COLLECTIONS_ELASTICSEARCH_URL
+
+pytestmark = [pytest.mark.django_db]
+
+
+def test_tags_api(fakedata, taskmanager, client, django_user_model):
+    root = fakedata.init()
+
+    testfile1 = (TESTDATA / './no-extension/file_pdf')
+    testfile2 = (TESTDATA / './disk-files/images/bikes.jpg')
+
+    with testfile1.open('rb') as f:
+        blob1 = fakedata.blob(f.read())
+
+    with testfile2.open('rb') as f:
+        blob2 = fakedata.blob(f.read())
+
+    fakedata.file(root, 'file1.pdf', blob1)
+    fakedata.file(root, 'file2.jpg', blob2)
+
+    taskmanager.run(limit=3000)
+
+    tags_mapping = {
+        blob1.md5: ['tag1', 'tag2'],
+        blob2.md5: ['tag1', 'tag3']
+    }
+    create_tags_csv(tags_mapping, '/tmp/tags.csv')
+    col = collections.current().name
+    user = django_user_model.objects.create_user(username='test', password='pw')
+    call_command('importtags', col=col, tags='/tmp/tags.csv', user=user.username, uuid='xyz-123', p=True)
+    digest1 = models.Digest.objects.get(blob_id=blob1.pk)
+    digest2 = models.Digest.objects.get(blob_id=blob2.pk)
+
+    assert models.DocumentUserTag.objects.filter(digest_id=digest1.pk, tag='tag1').exists()
+    assert models.DocumentUserTag.objects.filter(digest_id=digest1.pk, tag='tag2').exists()
+    assert models.DocumentUserTag.objects.filter(digest_id=digest2.pk, tag='tag1').exists()
+    assert models.DocumentUserTag.objects.filter(digest_id=digest2.pk, tag='tag3').exists()
+
+    # need to wait for the document to be indexed
+    time.sleep(5)
```

Comment:
this will make for flaky tests, that depend on machine load and other things we can't control.

Do a polling loop on the API for max. 2min, it will tell you when the tag has been indexed.